packages/trie/src/trie.ts

```diff
@@ -93,6 +96,8 @@ export class Trie {
         throw new Error('`valueEncoding` can only be set if a `db` is provided')
       }
       this._opts = { ...this._opts, ...opts }
+      this._opts.useKeyHashingFunction =
+        opts.common?.customCrypto.keccak256 ?? opts.useKeyHashingFunction ?? keccak256
```

Comment:
Just remember that recently in an exchange with @jochem-brouwer we stumbled upon a problem (unexpected evaluation) when using this kind of operator chaining, can't completely recall the constallation specifics though. 🤔

Did some basic tests and these went as expected:

<img width="414" alt="grafik" src="https://github.com/ethereumjs/ethereumjs-monorepo/assets/931137/eac78a81-f264-4e08-ad46-ba6be7ca388b">

So just dropping this here for awareness.

(maybe though it was something with the `?` operator from TypeScript? So a problem might occur when - in the case above - `common` is not defined? Maybe things would then not evaluate properly? Not sure if Jochem can give more insight here (If I recall correctly a solution for *our* problem were adding some brackets)

Reply:
I can't recall the specifics, do you maybe have some more context when we stumbled upon this? I do slightly remember a "gotcha" with these operators though, but I don't recall in which context :sweat_smile: